doc/design/csp.md

```diff
@@ -0,0 +1,96 @@
+# Design Doc: CSP in PaddlePaddle Fluid
+
+## Motivations
+
+Concurrent programming is important for deep learning.  Example applications include
+
+1. A thread uses the GPU for computing while the main thread keeps loading the next minibatch, and
```

Comment:
Maybe re-write to:
1.  The main thread keeps reading the next mini-batch while another thread uses the GPU for computing.
2.  The main thread performs the computation while another thread uploads the local gradients from each trainer to the parameter server.